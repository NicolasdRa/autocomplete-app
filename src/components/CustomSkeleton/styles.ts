import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme: Theme) =>
	createStyles({
		skeletonContainer: {
			width: '90vw',
			margin: '2rem auto',
			display: 'flex',
			flexDirection: 'column',
			justifyContent: 'start',
		},

		skeletonFilters: {
			transform: 'none',
			margin: '2rem 0 1rem 0',
			width: '100%',
			height: '8rem',
		},

		skeletonTable: {
			transform: 'none',
			margin: '2rem 0 1rem 0',
			width: '100%',
			height: '65vh',
		},
	}),
);
