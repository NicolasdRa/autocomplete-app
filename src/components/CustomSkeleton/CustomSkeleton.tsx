import { Box } from '@material-ui/core';
import { Skeleton } from '@material-ui/lab';

import { useStyles } from './styles';

export const CustomSkeleton = () => {
	const classes = useStyles();

	return (
		<Box className={classes.skeletonContainer}>
			<Skeleton className={classes.skeletonFilters} />
			<Skeleton className={classes.skeletonTable} />
		</Box>
	);
};
