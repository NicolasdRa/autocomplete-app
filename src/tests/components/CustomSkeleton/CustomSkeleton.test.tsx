import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { CustomSkeleton } from '../../../components/CustomSkeleton/CustomSkeleton';

describe('tests on <CustomSkeleton/>', () => {
	const wrapper = shallow(<CustomSkeleton />);

	test('should render <CustomSkeleton/> correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
