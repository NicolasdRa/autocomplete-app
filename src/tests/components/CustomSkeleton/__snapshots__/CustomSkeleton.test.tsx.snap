// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on <CustomSkeleton/> should render <CustomSkeleton/> correctly 1`] = `
<Styled(MuiBox)
  className="makeStyles-skeletonContainer-1"
>
  <WithStyles(ForwardRef(Skeleton))
    className="makeStyles-skeletonFilters-2"
  />
  <WithStyles(ForwardRef(Skeleton))
    className="makeStyles-skeletonTable-3"
  />
</Styled(MuiBox)>
`;
